src/rpdk/core/data_loaders.py

```diff
@@ -148,6 +141,37 @@ def load_resource_spec(resource_spec_file):  # noqa: C901
         LOG.debug("Resource spec validation failed", exc_info=True)
         raise SpecValidationError(str(e)) from e
 
+    if {
+        "maxresults",
+        "maxrecords",
+        "maxitems",
+        "nexttoken",
+        "nextmarker",
+        "nextpagetoken",
+        "pagetoken",
+        "paginationtoken",
+    } & set(map(str.lower, resource_spec.get("properties", []))):
+        LOG.warning(
+            "LIST API inputs like MaxResults, MaxRecords, MaxItems, NextToken, NextMarker, NextPageToken, PageToken, and Filters are not resource properties"
+        )
+
+    if set(resource_spec.get("readOnlyProperties", [])) & set(
+        resource_spec.get("createOnlyProperties", [])
+    ) or set(resource_spec.get("readOnlyProperties", [])) & set(
+        resource_spec.get("writeOnlyProperties", [])
+    ):
+        LOG.warning(
+            "readOnlyProperties cannot be specified by customers and should not overlap with writeOnlyProperties or createOnlyProperties"
+        )
```

Comment:
Is it possible for users to successfully submit their resources when these warning are logged? If yes, should we throw an exception and stop them from submitting?

Reply:
> Is it possible for users to successfully submit their resources when these warning are logged?

As it's currently written, yes

> If yes, should we throw an exception and stop them from submitting?

I've also been debating this. We should have enforced some of these from the start, but it's trickier now to suddenly introduce backwards incompatible breaking changes

Reply:
I think we should do it, maybe with a major version bump and a release announcement that the changes are backward incompatible. Users can stick with a specific version until they are ready to upgrade.

Reply:
i think this is something we can introduce in the future, but i think these warnings are still valuable for customers to see before upgrading. usually with some backwards incompatible change there's a period of warning about an upcoming change in behavior, then a flip of some kind. we can use this as an opportunity to introduce the warnings then remove the functionality at a later date

Reply:
I agree NOT to introduce backward incompatible change as because there are some resources have it's special cases which it will break them.

Warning is good at this point to alert.